File: Tarea 2/Articulo.php
<?php

/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/**
 * Description of Articulo
 *
 * @author Ernesto Ramirez
 */
class Articulo {
    
    public $sku;
    protected $existencia;
    protected $precio;
    protected $costo;
    public $descripción;
    
  public function Articulo ($sku, $precio, $descripcion) {
    $this->sku = $sku;
    $this->existencia = 0;
    $this->precio = $precio;
    $this->costo = 0;
    $this->descripción = $descripcion;
  }
    public function compra ($cArt, $costo) {
     $this->existencia += $cArt;
     $this->costo = $costo;
  }  
  public function verificarExistencia (){
        return $this->existencia;    
    }
    
     public function venta ($cArt) {
         if ($this->verificarExistencia() >= $cArt) {
            $this->existencia -= $cArt;
           return true;
       } else {
           return false;
       }
    }
    
    public function actualizaPrecio ($precio){
        $this->precio = $precio;
    }
}


 

File: Tarea 1/Articulo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package poo.eramirez.t01;

/**
 *
 * @author Netito
 */
public class Articulo {
    int sku;
    protected int existencia;
    protected double precio;
    protected double costo;
    String descripcion;
    
    public Articulo (int sku, double precio, String descripcion){
        this.sku = sku;
        this.existencia = 0;
        this.precio = precio;
        this.costo = 0;
        this.descripcion = descripcion;
    }
    
    public void compra (int cArt, double costo){
        this.existencia += cArt;
        this.costo = costo;  
    }
    
    public boolean venta (int cArt){
        if (verificarExistencia() >= cArt) {
            existencia -= cArt;
           return true;
       } else {
           return false;
       }
    }
    
    public int verificarExistencia (){
        return this.existencia;    
    }
    
    public void actualizaPrecio (double precio){
        this.precio = precio;
    }
}



